great_expectations/experimental/datasources/postgres_datasource.py

```diff
@@ -1,507 +1,7 @@
-                "table_name": self.table_name,
-                "batch_identifiers": {},
-            }
-            if column_splitter:
-                batch_spec_kwargs["splitter_method"] = column_splitter.method_name
-                batch_spec_kwargs["splitter_kwargs"] = {
-                    "column_name": column_splitter.column_name
-                }
-                # mypy infers that batch_spec_kwargs["batch_identifiers"] is a collection, but
-                # it is hardcoded to a dict above, so we cast it here.
-                cast(Dict, batch_spec_kwargs["batch_identifiers"]).update(
-                    {column_splitter.column_name: request.options}
-                )
-            batch_spec = SqlAlchemyDatasourceBatchSpec(**batch_spec_kwargs)
-            data, markers = self.datasource.execution_engine.get_batch_data_and_markers(
-                batch_spec=batch_spec
-            )
-
-            # batch_definition (along with batch_spec and markers) is only here to satisfy a
-            # legacy constraint when computing usage statistics in a validator. We hope to remove
-            # it in the future.
-            # imports are done inline to prevent a circular dependency with core/batch.py
-            from great_expectations.core import IDDict
-            from great_expectations.core.batch import BatchDefinition
-
-            batch_definition = BatchDefinition(
-                datasource_name=self.datasource.name,
-                data_connector_name="experimental",
-                data_asset_name=self.name,
-                batch_identifiers=IDDict(batch_spec["batch_identifiers"]),
-                batch_spec_passthrough=None,
-            )
-            batch_list.append(
-                Batch(
-                    datasource=self.datasource,
-                    data_asset=self,
-                    batch_request=request,
-                    data=data,
-                    metadata=batch_metadata,
-                    legacy_batch_markers=markers,
-                    legacy_batch_spec=batch_spec,
-                    legacy_batch_definition=batch_definition,
-                )
-            )
-        self._sort_batches(batch_list)
-        return batch_list
-
-
-class PostgresDatasource(Datasource):
-    """Postgres datasource
-
-    Args:
-        name: The name of this datasource
-        connection_str: The SQLAlchemy connection string used to connect to the database.
-            For example: "postgresql+psycopg2://postgres:@localhost/test_database"
-        assets: An optional dictionary whose keys are table asset names and whose values
-            are TableAsset objects.
-    """
-
-    # class var definitions
-    asset_types: ClassVar[List[Type[DataAsset]]] = [TableAsset]
-
-    # right side of the operator determines the type name
-    # left side enforces the names on instance creation
+class PostgresDatasource(SQLDatasource):
     type: Literal["postgres"] = "postgres"
-    connection_string: str
```

Comment:
Now that we have a general SQL datasource type.
Can we use the `PostgresDsn` type for `connection_string`?

https://docs.pydantic.dev/usage/types/#urls

```python
from pydantic import PostgresDsn

class PostgresDatasource(SQLDatasource):
    type: Literal["postgres"] = "postgres"
    connection_string: PostgresDsn
```